### PR TITLE
[Do not merge] AEROGEAR-2527 Spike ensuring device check approach

### PR DESCRIPTION
**Pull request by aidenkeating**
## Motivation

Just like in the other SDKs we need device checks.

## Description

Ensure that the approach of exposing other plugins within our SDK would work.

Note that this is a very simple approach just to confirm that everything works together, the final approach will be structured similar to the Android SDK checks.

### CODE REVIEW

**Comment by aidenkeating** [comment]
This will more than likely be replaced with an actual PR for rooted checks etc. This was the output of a spike to confirm the approach.

**Comment by josemigallas** [comment]
@aidenkeating Looks good and simple. I would only recommend consistency in names. IMO package (prefixed with @aerogear) and plugin should be the same.

@wtrocki we could add a new method in `MetricsService` that checks whether or no `IRoot` exists, something like `sendDeviceCheckMetrics`? Or some options passed in the constructor:
```
new MetricsService({ deviceCheck: true });
```

**Comment by wtrocki** [comment]
@josemigallas This should be initiated by other SDK created in this PR. Core should now know anything about security metrics.

File: packages/security/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+declare var IRoot: any;
+
+export default class SecurityService {
+  constructor() {}
+
+  public isRooted = IRoot.isRooted;
```
Comment (wei-lee) [inline packages/security/src/index.ts]: is it possible that `IRoot` is undefined?

Comment (wtrocki) [reply packages/security/src/index.ts]: Yeah. This code should use the same aproach as in core where we:

1) Detect platform
2) Detect if plugin is available.

**Comment by aidenkeating** [comment]
Closing this to be replaced with other, proper, PRs

**Pull request closed without merging**